run/github-runner/github-runner-autoscaler/main.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2025 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import hmac
+import hashlib
+import requests
+import json
+import logging
+from google.oauth2 import service_account
+from google.auth.transport.requests import Request as GoogleRequest
+import google.auth
+from flask import Request
+from google.cloud import secretmanager
+from github import Github
+
+
+# --- Configuration ---
+PROJECT_ID = os.environ.get('GCP_PROJECT')
+LOCATION = 'us-central1' # Or your desired region
```

Comment:
Can you retrieve this from metadata server?
See https://github.com/GoogleCloudPlatform/cloud-run-hello/blob/master/hello.go#L121